relay-server/src/utils/dynamic_sampling.rs

```diff
@@ -1,152 +1,228 @@
+#[derive(Debug, Copy, Clone, Serialize, Deserialize, Eq, PartialEq)]
 #[serde(rename_all = "camelCase")]
-pub enum SamplingStrategy {
-    Trace, // Rules that apply to Transaction items
-    Event, // Rules that apply to Event items
+pub enum RuleType {
+    /// A trace rule applies only to transactions and it is applied on the trace info
+    Trace,
+    /// A transaction rule applies to transactions and it is applied  on the transaction event
+    Transaction,
+    /// A non transaction rule applies to Errors, Security events...every type of event that
+    /// is not a Transaction
+    Error,
+}
+
+/// The value of a field extracted from a FieldValueProvider (Event or TraceContext)
+#[derive(Debug, Clone)]
+pub enum FieldValue<'a> {
+    String(&'a str),
+    None,
 }
 
-/// A sampling rule defined by user in Organization options.
 #[derive(Debug, Clone, Serialize, Deserialize)]
 #[serde(rename_all = "camelCase")]
-pub struct SamplingRule {
-    /// The project ids for which the sample rule applies, empty applies to all
-    #[serde(default)]
-    pub project_ids: Vec<ProjectId>,
-    /// A set of Glob patterns for which the rule applies, empty applies to all
-    #[serde(default)]
-    pub releases: GlobPatterns,
-    /// A set of user_segments for which the rule applies, empty applies to all
+pub struct EqCondData {
+    pub name: String,
+    pub value: Vec<String>,
     #[serde(default)]
-    pub user_segments: Vec<LowerCaseString>,
-    #[serde(default)]
-    pub environments: Vec<LowerCaseString>,
-    /// The sampling rate for trace matching this rule
-    pub sample_rate: f64,
-    /// Specifies to what type of item does this rule apply
-    pub strategy: SamplingStrategy,
+    pub ignore_case: bool,
 }
 
-#[derive(Debug, Clone, Serialize, Default)]
-pub struct LowerCaseString(String);
-
-impl LowerCaseString {
-    pub fn new(val: &str) -> LowerCaseString {
-        LowerCaseString(val.to_lowercase())
-    }
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct GlobCondData {
+    pub name: String,
+    pub value: GlobPatterns,
 }
 
-impl From<&str> for LowerCaseString {
-    fn from(val: &str) -> Self {
-        LowerCaseString(val.to_lowercase())
-    }
+/// Keeps inner conditions for combinator conditions
+/// This structure is used to aid the serialisation of Rules.
+/// Since we use internally tagged serialisation for RuleConditions
+/// We need to add an inner level when serializing in order not to have a
+/// clash fo the tag operator between the inner and the outer Conditions
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct InnerConditions {
+    inner: Vec<RuleCondition>,
 }
 
-impl PartialEq for LowerCaseString {
-    fn eq(&self, other: &Self) -> bool {
-        self.0 == other.0
-    }
+/// Keeps inner conditions for combinator conditions with one element (i.e. Not)
+/// This structure is used to aid the serialisation of Rules.
+/// See [InnerConditions] for further explanations.
```

Comment:
_nit_: Suggestions for improving doc formatting:

```suggestion
/// Negates a wrapped condition.
///
/// This structure is used to aid the serialization of Rules.
/// See [`InnerConditions`] for further explanations.
```

We could also link that this is used in `RuleCondition::Not`.